### ISSUE TITLE
Charts sizing problems with flexbox

### ISSUE THREAD

**Issue by datafoo** [amcharts/amcharts5#311]
### Steps to reproduce

1. Configure a chart within flexbox (see https://codepen.io/jackfoo/pen/rNYrpYd)
2. Increase the width of the viewport and notice how the charts take the new available space
2. Decrease the width of the viewport

### Expected result:
The charts width decreases due to less available space

### Actual result
The charts width stays the same and a windows slider appears

### Other weird behavior:
Uncomment line 20 of the CSS code and watch the charts grow automatically.

**Comment by sassomedia** [comment]
I don't think this is a chart issue. I tested this on the Mac in:
- Safari 15.3: scales as expected
- FireFox 94.0.2: doesn't scale but shows scrollbars
- Chrome 98.0.4758.109: doesn't scale but shows _no_ scrollbars
- Edge 98.0.1108.62: same as Chrome, it's the same engine underneath

Given that there are 3 different behaviors, this seems like a CSS issue. I would try to find another way to build your CSS setup.

**Comment by datafoo** [comment]
On my side (on Windows 10):

- Firefox 97.0.1
- Google Chrome 98.0.4758.102 and 99.0.4844.51
- Microsoft Edge 98.0.1108.62

They all act the same way: doesn't scale but shows scrollbar

It was working fine with amCharts 4 (I am trying to migrate to amCharts 5) and it works fine without a chart at all: https://codepen.io/jackfoo/pen/bGYOJgx

I do not believe that I am making invalid use of CSS so it seems the problem is on amCharts 5 side.

**Comment by datafoo** [comment]
> From your [demo](https://codepen.io/jackfoo/pen/rNYrpYd) though, I'm not sure why flexbox would be needed. Those are just two charts stacked on top of each other which could be done with two 100% divs. Is there something missing from that example?

Yes, I made the codepen as simple as I could to reproduce the problem.

**Comment by adybka-inovex** [comment]
I am having a similar problem where I have a I have a chart in a react bootstrap card which is in bootstrap Row and Col. Bootstrap rows and cols are written with flex box.
Where I have send sized for the app container based on window size. When it hits a breakpoint to shrink, the chart sizes become messed up and are no in one clean row. But in the reverse where a breakpoint is hit to grow the charts grow fine. There is a code sandbox I wrote that has this issues. Any help is really appreciated.

https://codesandbox.io/s/bootstrap-grid-card-chart-6rkvnz

**Comment by martynasma** [comment]
I suggest not setting `width: 100%` in a chart div CSS.

**Comment by datafoo** [comment]
> I suggest not setting `width: 100%` in a chart div CSS.

It does not solve the problem for me.

**Comment by martynasma** [comment]
It does solve problem in the CodePen you posted for me.

If you have different case, can you post it as well?

**Comment by datafoo** [comment]
> It does solve problem in the CodePen you posted for me.
> 
> If you have different case, can you post it as well?

So I just commented the CSS line 16 from my original CodePen and the result is: https://codepen.io/jackfoo/pen/LYePOXy

Here you can see the horizontal scrollbar:
![Screenshot 2022-03-11 at 14-49-04 Charts sizing problems with flexbox](https://user-images.githubusercontent.com/34766150/157880509-393fef49-44aa-48db-9764-498ea354e40b.png)

**Comment by martynasma** [comment]
Are you getting this when resizing the window?

**Comment by datafoo** [comment]
> Are you getting this when resizing the window?

Yes

To clarify, here is how it looks when I load the page:
![Screenshot 2022-03-14 at 08-45-28 Charts sizing problems with flexbox](https://user-images.githubusercontent.com/34766150/158127276-1068aa95-d03b-4699-8e7f-cf17992980a8.png)

Then, here is what happen when I decrease the width of the window, a horizontal scroll bar appear instead of the chart width adapting to the new width of the window:
![Screenshot 2022-03-14 at 08-45-43 Charts sizing problems with flexbox](https://user-images.githubusercontent.com/34766150/158127359-9efdfd16-4112-44e1-ba0f-c7e45e3fcafe.png)

When I increase the width more than the original one, then there is no problem. The chart width adapts to the new width of the window:
![Screenshot 2022-03-14 at 08-45-59 Charts sizing problems with flexbox](https://user-images.githubusercontent.com/34766150/158127673-1649a580-c595-485f-931c-e2cde05cebc2.png)

**Comment by Pauan** [comment]
@datafoo Thanks for the report, it's a bug in amCharts. We fixed it, the fix will be in the next release. We'll let you know when the next release happens.
